10-JavaScript/spread-and-rest-destructuring.md

```diff
@@ -0,0 +1,280 @@
+
+The `Math.max` function is a good example of a function you might choose to use the spread operator on, it takes numeric arguments (unlimited size) and returns the largest item.  
+
+```javascript
+const list = [3, 7, 25, 1, -50];
+
+const biggest = Math.max(...list);
+
+console.log(biggest); // 25
+```
+
+**Wait so these look the same!**  While these both involve the same three periods, examine where the `...` operator is placed.  
+
+- In rest parameters the `...` operator appears in the **function parameters** while in the example above, we used the spread operator to convert an array into individual items in the **function call**.  
+
+- The spread operator can only be applied to [iterable values](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols) like arrays and strings while rest parameters take in any kind of arguments and place them into an array.
+
+### Merging Arrays
+
+You can also use the spread operator to expand an array to a list of elements for a new array.
+
+```javascript
+const newStudent = 'Ada';
+
+const studentList = ['Charles', 'Grace', 'Katherine', 'Alexandra'];
+
+const newList = [newStudent, ...studentList];
+
+console.log(newList); // Ada,Charles,Grace,Katherine,Alexandra
+```
+
+**Challenge**  Use the spread operator to combine the two arrays below:
+
+```javascript
+const morningTracks = ['Master of Puppets', 'Black Metal', 'Paranoid'];
+const eveningTracks = ['Ella Mai', 'American Teen', '17'];
+
+// Merge the above arrays into a single array
+
+```
+
+<details>
+  <summary>
+    Click here to see an answer:
+  </summary>
+
+  `const combined = [...morningTracks, ...eveningTracks];`
+</details>
+
+### Spread operator with strings
+
+You can also break a string into an array of individual characters with the spread operator.
+
+```javascript
+const letters = [...'abc']; // ['a', 'b', 'c']
+```
+
+### Combining the spread operator with destructuring
+
+You can also use destructuring along with the spread operator to remove individual elements from an array.
+
+```javascript
+const list = ['peanut butter', 'chocolate', 'vanilla', 'cookies and cream'];
+
+const [first, second, ...rest] = list;
+
+console.log(first); // peanut butter
+console.log(second); // chocolate
+console.log(rest); // vanilla,cookies and cream 
+```
+
+### Spread operator with object literals
+
+The spread operator can also be used to combine or merge two object literals.  The code below merges two object literals.
+
+```javascript
+const ada = {
+  name: 'Ada Lovelace',
+  age: 103,
+};
+
+const language = {
+  name: 'C#',
+  about: 'A strongly typed language by Microsoft',
+};
+
+const merged = { ...ada, ...language };
+
+console.log(merged); // { name: 'C#', age: 103, about: 'A strongly typed language by Microsoft' }
+```
+
+Notice that it merges the objects in order, so that `ada`'s name is overwritten by that of `language`.  
+
```

Comment:
I would love a note/extra paragraph (here) that says something along the lines of:

Note: Using the spread operator to merge arrays or object literals gives you back an entirely new array or object that isn't the same as any of the originals. In some cases in program where you have to worry about [immutability](https://en.wikipedia.org/wiki/Immutable_object), this can be important.